### PR TITLE
Nano blade rewrite

### PR DESCRIPTION
**Pull request by WalshyDev**
## Short Description
Rewriting the way the nano blade works to be more inline with Tekkit/FTB.

## Additions/Changes/Removals
* Nano blade now does 20 base damage while on and 4 while off
* It utilises PDC
* Optimised in parts
* Energy now depletes while it is on rather than on hit

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

**Comment by J3fftw1** [comment]
When you are working on it anyways mind fixing the electric chestplate aswell

File: src/main/java/dev/j3fftw/litexpansion/Items.java [hunk 1]
```diff
@@ -461,6 +465,21 @@
...
+    public static final SlimefunItemStack FOOD_SYNTHESIZER = new SlimefunItemStack(
+        "FOOD_SYNTHESIZER",
+        new CustomItem(SkullItem.fromHash("a967efe969d264f635f2c201c34381ef59c72e16ec50af7692033121e22fba9c"),
+            "Food Synthesizer"),
+        "&dFood Synthesizer",
+        "",
+        "&fKeeps you fed with artificial food.",
+        "&fComes in the flavors!",
+        "",
+        "&c&o&8\u21E8 &e\u26A1 &70 / 100 J"
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/Items.java:477]: Food_synth.capacity?

File: src/main/java/dev/j3fftw/litexpansion/weapons/NanoBlade.java [hunk 2]
```diff
@@ -1,32 +1,56 @@
...
+    public static final float CAPACITY = 4_000;
+    public static final float PER_TICK_REMOVAL = 64;
+
     public NanoBlade() {
         super(Items.LITEXPANSION, Items.NANO_BLADE, MetalForge.RECIPE_TYPE, new ItemStack[] {
-                null, Items.MAG_THOR, null,
-                null, Items.MAG_THOR, null,
-                null, SlimefunItems.ADVANCED_CIRCUIT_BOARD, null
+                new ItemStack(Material.GLOWSTONE_DUST), Items.ADVANCED_ALLOY, null,
+                new ItemStack(Material.GLOWSTONE_DUST), Items.ADVANCED_ALLOY, null,
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/weapons/NanoBlade.java:33]: Is this the official recipe?

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/items/PassiveElectricRemoval.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    float getCapacity();
+
+    boolean isEnabled(@Nonnull ItemMeta meta);
+
+    static void tick(@Nonnull ItemStack is, @Nonnull PassiveElectricRemoval per) {
+        final ItemMeta meta = is.getItemMeta();
+
+        if (per.isEnabled(meta)) {
+//            final float charge = ChargeUtils.getCharge(meta);
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/items/PassiveElectricRemoval.java:37]: Remove comment

File: src/main/java/dev/j3fftw/litexpansion/items/PassiveElectricRemoval.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            final float charge = getCharge(meta);
+            if (charge == 0) return;
+
+            final float newCharge = Math.max(charge - per.getRemovedChargePerTick(), 0);
+
+//            ChargeUtils.setCharge(meta, newCharge, CAPACITY);
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/items/PassiveElectricRemoval.java:43]: Remove comment

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

**Comment by J3fftw1** [comment]
Also on top of the current requested changes if you dont mind can you remove cscorelib dependency and update bstats dependency

File: src/main/java/dev/j3fftw/litexpansion/items/PassiveElectricRemoval.java (hunk 3 above)
Comment (WalshyDev) [reply src/main/java/dev/j3fftw/litexpansion/items/PassiveElectricRemoval.java:37]: These are for the next RC

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/items/PassiveElectricRemoval.java:37]: Ah okay ignore my comment then

File: src/main/java/dev/j3fftw/litexpansion/Items.java (hunk 1 above)
Comment (WalshyDev) [reply src/main/java/dev/j3fftw/litexpansion/Items.java:477]: Not related

File: src/main/java/dev/j3fftw/litexpansion/weapons/NanoBlade.java (hunk 2 above)
Comment (WalshyDev) [reply src/main/java/dev/j3fftw/litexpansion/weapons/NanoBlade.java:33]: Yes - https://tekkitclassic.fandom.com/wiki/Nano_Saber

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/weapons/NanoBlade.java:33]: okay good

**Comment by J3fftw1** [comment]
> Also on top of the current requested changes if you dont mind can you remove cscorelib dependency and update bstats dependency

If you dont mind doing this then its ready for merge

**Comment by WalshyDev** [comment]
Alright, it's good to go

**Review by J3fftw1 — APPROVED** [review]
Lgtm

**Pull request merged**